File: services/Nemstore.Orders.Api/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Nemstore.Orders.Api.Models
{
    public class Order
    {
        public Guid Id { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public IList<OrderLine> Lines { get; set; }
        public OrderStatus Status { get; set; }
        public decimal Total => Lines.Sum(line => line.TotalPrice);
    }

    public class OrderLine
    {
        public Guid Id { get; set; }
        public Guid OrderId { get; set; }
        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public string ProductImageUrl { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal TotalPrice => UnitPrice * Quantity;
    }

    public enum OrderStatus 
    {
        Created, AwaitingStockApproval, Completed
    }
}

File: web/frontend/angular/nemstore-orders/src/app/models/Order.ts
export class Order {
    id: string;
    createdAt: string;
    lines: OrderLine[];
    status: OrderStatus;
    total: number;
    showOrderSummary: boolean = false;
}

export class OrderLine {
    id: string;
    orderId: string;
    productId: number;
    productName: string;
    productImageUrl: string;
    quantity: number;
    totalPrice: number;
}

export enum OrderStatus {
    Created, AwaitingStockApproval, Completed
}
